src/Native/System.Security.Cryptography.Native/pal_ssl.h

```diff
@@ -6,6 +6,154 @@
 #include <openssl/ssl.h>
 
 /*
+These values should be kept in sync with System.Security.Authentication.CipherAlgorithmType.
+*/
+enum class CipherAlgorithmType : int32_t
+{
+    None = 0,
+    Null = 24576,
+    Des = 26113,
+    Rc2 = 26114,
+    TripleDes = 26115,
+    Aes128 = 26126,
+    Aes192 = 26127,
+    Aes256 = 26128,
+    Aes = 26129,
+    Rc4 = 26625,
+
+    //Algorithm constants which are not present in the managed CipherAlgorithmType enum.
+    SSL_IDEA = 229380,
+    SSL_CAMELLIA128 = 229381,
+    SSL_CAMELLIA256 = 229382,
+    SSL_eGOST2814789CNT = 229383,
+    SSL_SEED = 229384,
+};
+
+/*
+The values used in OpenSSL for SSL_CIPHER algorithm_enc.
+*/
+enum class SSL_CipherAlgorithm : int64_t
+{
+#ifdef HAVE_SSL_CIPHER_SPLIT_ALGORITHMS
+    SSL_DES = 1,
+    SSL_3DES = 2,
+    SSL_RC4 = 4,
+    SSL_RC2 = 8,
+    SSL_IDEA = 16,
+    SSL_eNULL = 32,
+    SSL_AES128 = 64,
+    SSL_AES256 = 128,
+    SSL_CAMELLIA128 = 256,
+    SSL_CAMELLIA256 = 512,
+    SSL_eGOST2814789CNT = 1024,
+    SSL_SEED = 2048,
+    SSL_AES128GCM = 4096,
+    SSL_AES256GCM = 8192
+#else
```

Comment:
The declarations in the .h should be things that managed code can depend on unconditionally. There shouldn't be ifdefs changing enum values. If these are only used in the implementation and not expected to be known values to managed code, move them to the .cpp.